File: AD-HOC/1107/main.c
#include <stdio.h>

int numLasers(int a, int c, int x[c]) {
    int lasers = a - x[0];
    for (int i = 1; i < c; i++)
        if (x[i - 1] > x[i]) lasers += x[i - 1] - x[i];
    return lasers;
}

int main() {
    int a, c;
    for (scanf("%d %d", &a, &c); a || c; scanf("%d %d", &a, &c)) {
        int x[c];
        for (int i = 0; i < c; i++) scanf("%d", &x[i]);
        printf("%d\n", numLasers(a, c, x));
    }
    return 0;
}

File: AD-HOC/1107/main.rb
def num_lasers(a, c, x)
  lasers = a - x.first
  (1...c).each do |i|
    lasers += x[i - 1] - x[i] if x[i - 1] > x[i]
  end
  lasers
end

a, c = gets.split.map(&:to_i)
while a.nonzero? || c.nonzero?
  x = gets.split.map(&:to_i)
  puts num_lasers(a, c, x)
  a, c = gets.split.map(&:to_i)
end
